pkg/provider/kubernetes/gateway/kubernetes.go

```diff
@@ -0,0 +1,866 @@
+						// update "ResolvedRefs" status true with "DroppedRoutes" reason
+						listenerStatuses[i].Conditions = append(listenerStatuses[i].Conditions, metav1.Condition{
+							Type:               string(v1alpha1.ListenerConditionResolvedRefs),
+							Status:             metav1.ConditionFalse,
+							LastTransitionTime: metav1.Now(),
+							Reason:             string(v1alpha1.ListenerReasonDegradedRoutes),
+							Message:            msg,
+						})
+
+						logger.Errorf(msg)
+						// TODO update the RouteStatus condition / deduplicate conditions on listener
+						continue
+					}
+
+					if routeRule.ForwardTo != nil {
+						wrrService, subServices, err := loadServices(client, gateway.Namespace, routeRule.ForwardTo)
+						if err != nil {
+							msg := fmt.Sprintf("Cannot load service from HTTPRoute %s/%s : %v", gateway.Namespace, httpRoute.Name, err)
+							// update "ResolvedRefs" status true with "DroppedRoutes" reason
+							listenerStatuses[i].Conditions = append(listenerStatuses[i].Conditions, metav1.Condition{
+								Type:               string(v1alpha1.ListenerConditionResolvedRefs),
+								Status:             metav1.ConditionFalse,
+								LastTransitionTime: metav1.Now(),
+								Reason:             string(v1alpha1.ListenerReasonDegradedRoutes),
+								Message:            msg,
+							})
+
+							logger.Errorf(msg)
+							// TODO update the RouteStatus condition / deduplicate conditions on listener
+							continue
+						}
+
+						for svcName, svc := range subServices {
+							services[svcName] = svc
+						}
+
+						serviceName := provider.Normalize(routerKey + "-wrr")
+						services[serviceName] = wrrService
+
+						router.Service = serviceName
+					}
+
+					if router.Service != "" {
+						routerKey = provider.Normalize(routerKey)
+						routers[routerKey] = &router
+					}
+				}
+			}
+
+			if len(listenerStatuses[i].Conditions) == 0 {
+				// GatewayConditionReady "Ready", GatewayConditionReason "ListenerReady"
+				listenerStatuses[i].Conditions = append(listenerStatuses[i].Conditions, metav1.Condition{
+					Type:               string(v1alpha1.ListenerConditionReady),
+					Status:             metav1.ConditionTrue,
+					LastTransitionTime: metav1.Now(),
+				})
+			} else {
+				listenerInError = true
+			}
+		}
+
+		gatewayStatus.Listeners = listenerStatuses
+
+		if listenerInError {
+			// GatewayConditionReady "Ready", GatewayConditionReason "ListenersNotValid"
+			gatewayStatus.Conditions = append(gatewayStatus.Conditions, metav1.Condition{
+				Type:               string(v1alpha1.GatewayConditionReady),
+				Status:             metav1.ConditionFalse,
+				LastTransitionTime: metav1.Now(),
+				Reason:             string(v1alpha1.GatewayReasonListenersNotValid),
+				Message:            "All Listeners must be valid",
+			})
+
+			err := client.UpdateGatewayStatus(gateway, gatewayStatus)
+			if err != nil {
+				logger.Debugf("An error occurred while updating gateway status: %v", err)
```

Comment:
I think it's better to log an error here, wdyt?

```suggestion
				logger.Errorf("An error occurred while updating gateway status: %v", err)
```